src/Core/src/Platform/Windows/NavigationManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Media.Animation;
+using Microsoft.UI.Xaml.Navigation;
+
+namespace Microsoft.Maui
+{
+	public class NavigationManager
+	{
+		IView? _currentPage;
+		IMauiContext _mauiContext;
+		NavigationFrame? _navigationFrame;
+		protected WindowManager WindowManager => _mauiContext.GetWindowManager();
+		private protected INavigationView? NavigationView { get; private set; }
+		public IReadOnlyList<IView> NavigationStack { get; set; } = new List<IView>();
+		public IMauiContext MauiContext => _mauiContext;
+		public IView CurrentPage
+			=> _currentPage ?? throw new InvalidOperationException("CurrentPage cannot be null");
+		public NavigationFrame NavigationFrame =>
+			_navigationFrame ?? throw new InvalidOperationException("NavigationFrame Null");
+
+		public NavigationManager(IMauiContext mauiContext)
+		{
+			_mauiContext = mauiContext;
+		}
+
+		public virtual void Connect(IView navigationView, NavigationFrame navigationFrame)
```

Comment:
Don't we wan't a DIsconnect to unsubscribe `_navigationFrame.Navigated `

Reply:
👍